cmake/CMakeLists.txt

```diff
@@ -49,6 +49,11 @@ if (${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
   set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -restrict")
 endif()
 
+option(ENABLE_COVERAGE "Enable code coverage" OFF)
+if(ENABLE_COVERAGE)
+  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage")
```

Comment:
Doesn't it need to go in `CMAKE_EXE_LINKER_FLAGS` and `CMAKE_SHARED_LINKER_FLAGS` as well?

Reply:
apparently not (output of `make VERBOSE=1`):
```
/usr/bin/c++   --coverage -fopenmp -O3 -DNDEBUG  -rdynamic CMakeFiles/lmp.dir/home/richard/GitHub/rbberger/lammps/src/main.cpp.o  -o lmp -Wl,-rpath,/usr/lib/openmpi: liblammps.a /usr/lib/openmpi/libmpi_cxx.so /usr/lib/openmpi/libmpi.so -ljpeg -lpng -lz -lm 
```